no font anti-aliasing

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Run Zed Dev 0.142.0, and Zed Preview 0.144.0.
The UI and buffer fonts in 0.142.0 look nice and smooth.
Observe the UI and buffer fonts in 0.144.0, which are sharp and not anti-aliased.

### Environment

(text looks pretty and anti-aliased in UI and buffer)
Zed: v0.142.0 (Zed Dev 3ee3c6a3bd77370de952831b1f99b7dc026b2b95)
OS: Linux Wayland ubuntu 24.04
Memory: 31 GiB
Architecture: x86_64

(ugly text in UI and buffer no anti-aliasing)
Zed: v0.144.0 (Zed Preview)
OS: Linux Wayland ubuntu 24.04
Memory: 31 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

**Zed Preview 0.144.0**
![Screenshot from 2024-07-11 10-54-15](https://github.com/zed-industries/zed/assets/16313226/2f44ae1c-679a-4473-ac46-6459f274b8c0)

**Zed Dev 0.142.0**
![Screenshot from 2024-07-11 10-53-23](https://github.com/zed-industries/zed/assets/16313226/90de6591-d34e-42f0-9f7c-eb372d837220)

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-07-11T10:56:03+02:00 [INFO] ========== starting zed ==========
2024-07-11T10:56:03+02:00 [INFO] Opening main db
2024-07-11T10:56:03+02:00 [INFO] perform;
2024-07-11T10:56:03+02:00 [INFO] read_command;
2024-07-11T10:56:03+02:00 [INFO] read_command;
2024-07-11T10:56:03+02:00 [INFO] socket reader;
2024-07-11T10:56:03+02:00 [INFO] new;
2024-07-11T10:56:03+02:00 [INFO] Using git binary path: None
2024-07-11T10:56:03+02:00 [INFO] keep_updated;
2024-07-11T10:56:03+02:00 [INFO] extensions updated. loading 5, reloading 0, unloading 0
2024-07-11T10:56:03+02:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-07-11T10:56:03+02:00 [INFO] Opening main db
2024-07-11T10:56:03+02:00 [INFO] perform;
2024-07-11T10:56:03+02:00 [INFO] read_command;
2024-07-11T10:56:03+02:00 [INFO] read_command;
2024-07-11T10:56:03+02:00 [INFO] socket reader;
2024-07-11T10:56:03+02:00 [INFO] set environment variables from shell:/bin/bash, path:/home/ciaran/.opam/smtcoqcapi/bin:/home/ciaran/.local/bin:/home/ciaran/.opam/smtcoqcapi/bin:/home/ciaran/.cargo/bin:/home/ciaran/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
2024-07-11T10:56:03+02:00 [INFO] Enabling Vulkan Portability
2024-07-11T10:56:03+02:00 [INFO] Enabling color space support
2024-07-11T10:56:03+02:00 [INFO] Adapter "NVIDIA RTX A1000 6GB Laptop GPU"
2024-07-11T10:56:03+02:00 [INFO] Ray tracing is supported
2024-07-11T10:56:03+02:00 [INFO] Using surface present mode MAILBOX
2024-07-11T10:56:03+02:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T10:56:03+02:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T10:56:03+02:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T10:56:03+02:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-07-11T10:56:03+02:00 [INFO] Opening main db
2024-07-11T10:56:03+02:00 [INFO] Using surface present mode MAILBOX
2024-07-11T10:56:03+02:00 [WARN] Unable to forbid exclusive full screen
2024-07-11T10:56:03+02:00 [INFO] Opening main db
2024-07-11T10:56:03+02:00 [ERROR] No path stored for this editor
2024-07-11T10:56:04+02:00 [INFO] Node runtime install_if_needed
2024-07-11T10:56:04+02:00 [INFO] starting language server. binary path: "/home/ciaran/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/", args: ["/home/ciaran/.local/share/zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-07-11T10:56:04+02:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-07-11T08:56:04.763Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-07-11T08:56:04.763Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-07-11T10:56:04+02:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "26df1674-92c6-408e-864a-d1cf948c52b1",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-07-11T10:56:04+02:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-07-11T08:56:04.774Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-07-11T08:56:04.774Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-07-11T10:56:06+02:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T10:56:06+02:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-07-11T10:56:06+02:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-07-11T10:56:06+02:00 [INFO] Creating a descriptor pool for at most 256 sets
<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/gpui/src/platform/linux/text_system.rs
+++ b/crates/gpui/src/platform/linux/text_system.rs
@@ -1,7 +1,7 @@
 use crate::{
     point, size, Bounds, DevicePixels, Font, FontFeatures, FontId, FontMetrics, FontRun, FontStyle,
     FontWeight, GlyphId, LineLayout, Pixels, PlatformTextSystem, Point, RenderGlyphParams,
-    ShapedGlyph, SharedString, Size,
+    ShapedGlyph, SharedString, Size, SUBPIXEL_VARIANTS,
 };
 use anyhow::{anyhow, Context, Ok, Result};
 use collections::HashMap;
@@ -284,11 +284,10 @@ impl CosmicTextSystemState {
 
     fn raster_bounds(&mut self, params: &RenderGlyphParams) -> Result<Bounds<DevicePixels>> {
         let font = &self.loaded_fonts_store[params.font_id.0];
-        let font_system = &mut self.font_system;
         let image = self
             .swash_cache
             .get_image(
-                font_system,
+                &mut self.font_system,
                 CacheKey::new(
                     font.id(),
                     params.glyph_id.0 as u16,
@@ -315,19 +314,20 @@ impl CosmicTextSystemState {
         if glyph_bounds.size.width.0 == 0 || glyph_bounds.size.height.0 == 0 {
             Err(anyhow!("glyph bounds are empty"))
         } else {
-            // todo(linux) handle subpixel variants
             let bitmap_size = glyph_bounds.size;
             let font = &self.loaded_fonts_store[params.font_id.0];
-            let font_system = &mut self.font_system;
+            let subpixel_shift = params
+                .subpixel_variant
+                .map(|v| v as f32 / (SUBPIXEL_VARIANTS as f32 * params.scale_factor));
             let mut image = self
                 .swash_cache
                 .get_image(
-                    font_system,
+                    &mut self.font_system,
                     CacheKey::new(
                         font.id(),
                         params.glyph_id.0 as u16,
                         (params.font_size * params.scale_factor).into(),
-                        (0.0, 0.0),
+                        (subpixel_shift.x, subpixel_shift.y.trunc()),
                         cosmic_text::CacheKeyFlags::empty(),
                     )
                     .0,

